File: set_webhook.py
import os
import telebot
from dotenv import load_dotenv

load_dotenv()

BOT_TOKEN = os.getenv("BOT_TOKEN")
if not BOT_TOKEN:
    raise ValueError("No BOT_TOKEN provided. Please set it in your .env file.")

bot = telebot.TeleBot(BOT_TOKEN)

# Replace this with your actual Vercel URL
VERCEL_URL = "https://your-vercel-project.vercel.app/api"

# Remove any previous webhook
bot.remove_webhook()

# Set the new webhook
bot.set_webhook(url=VERCEL_URL)

print(f"Webhook set successfully to {VERCEL_URL}")

File: scripts/set-webhook.js
require("dotenv").config();

const TOKEN = process.env.BOT_TOKEN;
const URL = process.env.WEBHOOK_URL;

if (!TOKEN || !URL) {
  console.error("Set BOT_TOKEN and WEBHOOK_URL in .env");
  console.error("WEBHOOK_URL example: https://your-app.vercel.app/api/webhook");
  process.exit(1);
}

fetch(`https://api.telegram.org/bot${TOKEN}/setWebhook?url=${encodeURIComponent(URL)}`)
  .then((r) => r.json())
  .then((data) => console.log(data))
  .catch((err) => console.error(err));
